metrics/internal/cache/pv.go

```diff
@@ -167,7 +171,7 @@ func (p *PersistentVolumeStore) Add(obj interface{}) error {
 
 func getNodeNameForPV(pv *corev1.PersistentVolume, kubeClient clientset.Interface) (string, error) {
 	if pv.Spec.ClaimRef == nil {
-		return "", fmt.Errorf("persistent volume %s is not bound to any claim", pv.Name)
+		return "", nil
```

Comment:
You can have PV with claimRef which is some PVC already deleted (doesn't exists anymore) but you can have status.Phase =Release but not Bound.


```yaml
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f rbd $]kubectl get pvc
No resources found in default namespace.
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f rbd $]kubectl get pv
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS     CLAIM             STORAGECLASS      REASON   AGE
pvc-b00344cb-9c54-4596-bde3-e19447fecf75   1Gi        RWO            Retain           Released   default/rbd-pvc   rook-ceph-block            7m10s
[🎩︎]mrajanna@li-2cfbef4c-22d9-11b2-a85c-a3e4a93c405f rbd $]kubectl get pv -oyaml
apiVersion: v1
items:
- apiVersion: v1
  kind: PersistentVolume
  metadata:
    annotations:
      pv.kubernetes.io/provisioned-by: rook-ceph.rbd.csi.ceph.com
      volume.kubernetes.io/provisioner-deletion-secret-name: rook-csi-rbd-provisioner
      volume.kubernetes.io/provisioner-deletion-secret-namespace: rook-ceph
    creationTimestamp: "2024-01-22T16:42:32Z"
    finalizers:
    - kubernetes.io/pv-protection
    name: pvc-b00344cb-9c54-4596-bde3-e19447fecf75
    resourceVersion: "10934"
    uid: 1d10ed77-3cc9-40dc-9a87-350aa80b422e
  spec:
    accessModes:
    - ReadWriteOnce
    capacity:
      storage: 1Gi
    claimRef:
      apiVersion: v1
      kind: PersistentVolumeClaim
      name: rbd-pvc
      namespace: default
      resourceVersion: "10687"
      uid: b00344cb-9c54-4596-bde3-e19447fecf75
    csi:
      controllerExpandSecretRef:
        name: rook-csi-rbd-provisioner
        namespace: rook-ceph
      driver: rook-ceph.rbd.csi.ceph.com
      fsType: ext4
      nodeStageSecretRef:
        name: rook-csi-rbd-node
        namespace: rook-ceph
      volumeAttributes:
        clusterID: rook-ceph
        imageFeatures: layering
        imageFormat: "2"
        imageName: csi-vol-511a8008-418d-4423-8b31-6c8a8418a896
        journalPool: replicapool
        pool: replicapool
        storage.kubernetes.io/csiProvisionerIdentity: 1705941608727-3282-rook-ceph.rbd.csi.ceph.com
      volumeHandle: 0001-0009-rook-ceph-0000000000000001-511a8008-418d-4423-8b31-6c8a8418a896
    persistentVolumeReclaimPolicy: Retain
    storageClassName: rook-ceph-block
    volumeMode: Filesystem
  status:
    phase: Released
kind: List
metadata:
  resourceVersion: ""
```

Reply:
Ah didn't think about that. We must handle that case before listing PVCs.
Good catch again.